compiler/rustc_errors/src/json.rs

```diff
@@ -458,22 +458,33 @@ impl DiagnosticSpan {
         mut backtrace: impl Iterator<Item = ExpnData>,
         je: &JsonEmitter,
     ) -> DiagnosticSpan {
-        let start = je.sm.lookup_char_pos(span.lo());
+        let empty_source_map;
+        let sm = match &je.sm {
+            Some(s) => s,
+            None => {
+                span = rustc_span::DUMMY_SP;
+                empty_source_map = Arc::new(SourceMap::new(FilePathMapping::empty()));
+                empty_source_map
+                    .new_source_file(std::path::PathBuf::from("empty.rs").into(), String::new());
```

Comment:
Is it actually necessary to add a file to the source map? Or is an empty source map + DUMMY_SP enough?

Reply:
It's necessary to add a file. Otherwise it panics when trying to look up the span, with the same error as before.